File: app/src/androidTest/java/daotests/FavoriteTweetDaoTest.java
package daotests;

import org.junit.After;
import org.junit.AfterClass;
import org.junit.Before;
import org.junit.BeforeClass;
import org.junit.Test;

public class FavoriteTweetDaoTest extends DaoTest {


    @BeforeClass
    public static void initDatabase() {

    }

    @Test
    public void insertFavoriteTweet() {

    }

    @Test
    public void insertFavoriteTweetFails_ConflictTest() {

    }

    @Test
    public void insertFavoriteTweet_DuplicateTest() {

    }


    @Test
    public void insertFavoriteTweets() {

    }

    @Test
    public void insertFavoriteTweets_WithConflicts() {

    }

    @Test
    public void deleteFavoriteTweet_WithFavoriteTweet() {

    }

    @Test
    public void deleteFavoriteTweet_WithTweetId() {

    }

    @Test
    public void deleteAllFavoriteTweets() {

    }

    @Test
    public void getFavoriteTweetsCursor() {

    }

    @Test
    public void getLatestFavoriteTweetIds(int account) {

    }


    @Test
    public void tweetExists(long tweetId, int account) {

    }

    @Test
    public void trimDatabaseTrimsDatabase(int account, int trimSize) {

    }

    @After
    public void clearTables() {
        testDatabase.clearAllTables();
    }

    @AfterClass
    public static void closeDatabase() {
        testDatabase.close();
        testDatabase = null;
    }
}

File: app/src/androidTest/java/daotests/FavoriteTweetDaoTest.kt
package daotests

import android.content.ContentValues
import android.content.Context
import android.database.sqlite.SQLiteDatabase
import androidx.room.Query
import androidx.test.platform.app.InstrumentationRegistry
import com.klinker.android.twitter_l.data.roomdb.entities.FavoriteTweet
import com.klinker.android.twitter_l.data.roomdb.entities.Tweet
import com.klinker.android.twitter_l.data.roomdb.entities.User
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.*
import org.junit.After
import org.junit.AfterClass
import org.junit.Before
import org.junit.BeforeClass
import org.junit.Test

class FavoriteTweetDaoTest : DaoTest() {

    private val favoriteTweetDao get() = testDatabase.favoriteTweetDao()


    @After
    fun clearTables() {
        clearTestDatabase()
    }


    companion object {


        @BeforeClass
        @JvmStatic fun initDatabase() {
            DaoTest.initTestDatabase()
        }

        @AfterClass
        @JvmStatic fun closeDatabase() {
            DaoTest.closeTestDatabase()
        }

    }
}
